Vote on a review

PeterM24 opened the pull request:
Preview:
![Vote buttons](https://user-images.githubusercontent.com/123666879/233337470-24b8ba08-2371-45d0-ab4f-039485e0a101.gif)

Offline handling:
![Offline voting](https://user-images.githubusercontent.com/123666879/233336314-30b72759-0d66-4cf5-89e6-9060e6abbd58.gif)

PeterM24 committed:
Create basic voting buttons component

PeterM24 committed:
Move child key for comment list

PeterM24 committed:
initial logic for voting component

PeterM24 committed:
create patchVotes api util

PeterM24 committed:
fix patchVotes api util:
- fix path
- fix second axios arg

PeterM24 committed:
Complete functionality of voting component:
- Vote up can only be done once
- Vote down can only be done once
- Total votes updates optimistically
- Buttons disable
- UX message displays telling user success

PeterM24 committed:
Remove vote count from parent component

PeterM24 committed:
Update ReviewVote.tsx:
- handle undo once voted
- simplify logic
- change in user response messages

PeterM24 committed:
Link header to home Route
- links to all reviews

PeterM24 committed:
Update single review page:
- rearrange divs to use CSS grid
- remove comment count now that comment feature is added

PeterM24 committed:
Update CSS:
- changes made to single review page
- new vote button styling

PeterM24 committed:
Refactor ReviewVote.tsx:
- simplify button content using ternaries

PeterM24 committed:
Update review vote CSS

PeterM24 committed:
- extract ReviewVotesProps
- format date on single review page

popatre commented on src/components/ReviewVote.tsx:
```diff
@@ -0,0 +1,60 @@
+  const [hasVotedUp, setHasVotedUp] = useState<boolean>(false);
+  const [hasVotedDown, setHasVotedDown] = useState<boolean>(false);
+
+  const handleVoteUp = async () => {
+    setVotes((prevVotes) => prevVotes + 1);
+    setHasVotedUp(true);
+    await patchVotes(review_id, 1);
+  };
+
+  const handleVoteDown = async () => {
+    setVotes((prevVotes) => prevVotes - 1);
+    setHasVotedDown(true);
+    await patchVotes(review_id, -1);
+  };
```
Nice optimistic rendering - however, what happens if I dont have an internet connection/the request fails and click the button?

At the moment, it looks like all has gone well, until you refresh the page. 

Could you handle this occasion and make sure theres some feedback to let us know something has gone wrong?

PeterM24 replied:
@popatre Cheers! I've updated now - thanks for the feedback :)

popatre reviewed (changes requested):
Nice work here. However, just an issue when you dont have an internet connection and try to click the button - see my comment for more details

PeterM24 committed:
Handle offline voting:
- useState error msg updates when catch-block entered
- vote-header updates to errorMsg string
- additional button conditions add "Try again" within button

PeterM24 committed:
Remove focus border when buttons are disabled

PeterM24 committed:
Add specific error useStates for voting up and down
- stops both voting buttons displaying "Try again"

PeterM24 committed:
Changes made to handleUndo to handle offline:
- Vote count resets to initial state to avoid confusing UX
